File: python_server/ExecutedContract.py
class ExecutedContract:

    def __init__(self, contract_id, product, user, subordinate, client, date_of_payment, base_value, number_of_points, in_eur, paid_out):
        self.contract_id = contract_id
        self.product = product
        self.user = user
        self.subordinate = subordinate
        self.client = client
        self.date_of_payment = date_of_payment
        self.base_value = float(base_value)
        self.number_of_points = float(number_of_points)
        self.in_eur = float(in_eur)
        self.paid_out = paid_out

    def encode(self):
        return self.__dict__

File: angular/src/app/executed-contract.ts
import { Client } from "./client"
import { Product } from "./product"
import { User } from "./user"

export interface ExecutedContract {
    contract_id: number;
    product: Product;
    user: User;
    subordinate: User;
    client: Client
    date_of_payment: Date;
    base_value: number;
    number_of_points: number;
    in_eur: number;
    paid_out: string;
}
